mypyc/genops.py

```diff
@@ -3229,7 +3226,13 @@ def instantiate_callable_class(self, fn_info: FuncInfo) -> Value:
 
         # Set the callable class' environment attribute to point at the environment class
         # defined in the callable class' immediate outer scope.
-        self.add(SetAttr(func_reg, ENV_ATTR_NAME, self.fn_info.curr_env_reg, fitem.line))
+        if self.fn_info.is_generator:
+            curr_env_reg = self.fn_info.generator_class.curr_env_reg
+        elif self.fn_info.is_nested:
+            curr_env_reg = self.fn_info.callable_class.curr_env_reg
+        else:
+            curr_env_reg = self.fn_info.curr_env_reg
```

Comment:
Could you add some docs (not necessarily in this function) explaining the relationship between all the different `curr_env_reg`s and how they interact?

Reply:
Done.